data_transfer/jobs/shared.py

```diff
@@ -26,3 +45,68 @@ def batch_upload_data() -> None:
                 record.is_uploaded = True
                 update_record(record)
             dmpy.rm_local_data(zip_path)
+
+
+def prepare_data_folders() -> None:
+    """
+    Checks folders present in 'data/upload/ are finished
+    and moves them into the upload folder in the format:
+
+        DEVICEID-PATIENTID-STARTWEAR-ENDWEAR
+    """
+
+    @dataclass
+    class taskObject:
+        patient_id: str
+        device_id: str
+        folder: Path
+        records: List[Record]
+
+    ignore_folders = unfinished_folders()
+    all_folders = [
+        (p.name, d.name)
+        for p in config.storage_vol.iterdir()
+        if p.is_dir()
+        for d in p.iterdir()
+        if d.is_dir()
+    ]
+    ready_folders = [f for f in all_folders if f not in ignore_folders]
+
+    to_process = [
+        taskObject(
+            folder[0],
+            folder[1],
+            (config.storage_vol / folder[0] / folder[1]),
+            get_records_in_folder(folder[0], folder[1]),
+        )
+        for folder in ready_folders
+    ]
+
+    for task in to_process:
+        max_data, min_data = min_max_data_wear_times(task.records)
+
+        start_data = max_data.strftime("%Y%m%d")
+        end_data = min_data.strftime("%Y%m%d")
+
+        data_folder_name = (
+            f"{task.patient_id}-{task.device_id.replace('-', '')}"
+            f"-{start_data}-{end_data}"
+        )
+        destination = config.upload_folder / data_folder_name
+
+        if not config.upload_folder.exists():
+            config.upload_folder.mkdir()
+
+        if not destination.exists():
+            destination.mkdir()
+
+        # assuming all downloads went well: move everything and update records
+        for fname in task.folder.glob("*.*"):  # grabs all files
+
+            new_path = destination / fname.name
+            fname.rename(new_path)
```

Comment:
This also means you can remove this code, since we should move the complete folder (and all its contents) from `/input/PatientID/DeviceID` to `/upload/{destination}`